smartfs.lua

```diff
@@ -295,6 +321,55 @@ local function _sfs_recieve_(state,name,fields)
 	return true
 end
 
+-- Receive input from sender to the node form
+function smartfs.nodemeta_on_receive_fields(nodepos, formname, fields, sender)
+
+	-- get form info and check if it's a smartfs one
+	local meta = minetest.env:get_meta(nodepos)
+	local nodeform = meta:get_string("smartfs_name")
+	if not nodeform then -- execute only if it is smartfs form
+		print("SmartFS - (Warning) smartfs.nodemeta_on_receive_fields for node without smarfs data")
+		return false
+	end
+
+	-- get the currentsmartfs state
+	local opened_id = minetest.pos_to_string(nodepos)
+	local state = nil
+	local form = smartfs:__call(nodeform)
+	if not smartfs.opened[opened_id] or      --if opened first time
+	       smartfs.opened[opened_id].def.name ~= nodeform then --or form is changed
+		local params = minetest.deserialize(meta:get_string("smartfs_param")) -- param always persist between calls
+		state = smartfs._makeState_(form, sender, params, nil, nodepos)
+		smartfs.opened[opened_id] = state
+		form._reg(state)
+	else
+		state = smartfs.opened[opened_id]
+		state.player = sender -- set current user (temporary for this call)
+	end
+
+	-- Set current sender check for multiple users on node
+	state.multiplayer[sender:get_player_name()] = sender
+
+	-- take the input
+	_sfs_recieve_(state,opened_id,fields)
+
+	--persist parameter for later usage
+	meta:set_string("smartfs_param", minetest.serialize(state.param))
+	if state.closed then -- current user leave the node
+		state.multiplayer[sender:get_player_name()] = nil
+		if next(state.multiplayer) == nil then -- multiplayer table empty
+			--update formspec on node to a initial one for the next usage (respecting param persistence
+			state._ele = {} --reset the form
```

Comment:
maybe add state:clear()? Would be nice as a public API, and there may be more things to clear in the future